link/tracepoint_test.go

```diff
@@ -75,3 +76,47 @@ func TestTraceGetEventID(t *testing.T) {
 		t.Fatal("Doesn't return ErrNotSupported")
 	}
 }
+
+func TestTracepointProgramCall(t *testing.T) {
+	// Kernels before 4.14 don't support attaching to syscall tracepoints.
+	testutils.SkipOnOldKernel(t, "4.14", "syscalls tracepoint support")
+
+	// Create ebpf map. Will contain only one key with initial value 0.
+	m, err := ebpf.NewMap(&ebpf.MapSpec{
+		Type:       ebpf.Array,
+		KeySize:    4,
+		ValueSize:  4,
+		MaxEntries: 1,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create ebpf program. When called, will set the value of key 0 in
+	// the map created above to 1.
+	p := newMapUpdaterProg(t, m, ebpf.TracePoint)
+
+	// Open Tracepoint at /sys/kernel/debug/tracing/events/syscalls/sys_enter_getpid
+	// and attach it to the ebpf program created above.
+	tp, err := Tracepoint("syscalls", "sys_enter_getpid", p)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func(l Link) {
+		if err := l.Close(); err != nil {
```

Comment:
This makes me realise that testing the behaviour of Close() might be worth it?
* Always defer tp.Close()
* At the end of the test call tp.Close() and check that err == nil
* Then check that calling os.Getpid again doesn't update the map any more

Similar tests can be added for kprobe / uprobe.

Reply:
Good idea! To do this I would need to modify the ebpf program (currently I just set the value '1', I don't increment it)

Reply:
Maybe easier: just reset the value to zero before running the test. Keeping a counter has problems: for the Uprobe example we could end up in a state where something unrelated on the system executes /bin/bash during the test and we'd get a flaky failure.

Reply:
Yes, I'll do this way, I set the value in the first place in order to not depend on external events triggering the update